src/System.Linq.Expressions/src/System/Linq/Expressions/Interpreter/LightCompiler.cs

```diff
@@ -2790,7 +2790,11 @@ private void CompileTypeEqualExpression(Expression expr)
             var node = (TypeBinaryExpression)expr;
 
             Compile(node.Expression);
-            if (node.TypeOperand.GetTypeInfo().IsGenericType && node.TypeOperand.GetGenericTypeDefinition() == typeof(Nullable<>))
+            if (node.Expression.Type == typeof(void))
+            {
+                _instructions.EmitLoad(node.TypeOperand == node.Expression.Type, typeof(bool));
```

Comment:
Let's await the outcome of the discussion of issue https://github.com/dotnet/corefx/issues/5244 to decide on whether the change is warranted here, but obviously behavior should be consistent across compiler and interpreter no matter what we decide.